pkg/analytics/analytics.go

```diff
@@ -113,11 +113,12 @@ var (
 	// sensitiveFieldNames is a list of field names that should be sanitized.
 	// data sanitization is used to prevent sensitive data from being sent to the analytics server.
 	sensitiveFieldNames []string = []string{
-		"tfc-token",
-		"azurerm-account-key",
-		"fetch-tfstate-headers",
-		"username",
-		"password",
+		"headers",
```

Comment:
Is this meant to be an exhaustive list of all field names that could be potentially sensitive? If yes, it's practically infeasible to predict what kind of arguments could be sensitive. 

Why not to sanitise all arguments that are not registered with the engine instead, as a potential solution?

Reply:
Good point, instead of blacklist, use a whitelist.

In general for CMD arguments that might be possible, even if we currently don't have a list of all known/supported args for the legacy CLI. CMD arguments are just one example, the sanitization function works on a more abstract level, the json data structure that is send. Analytics have additional fields that can be filled by Extensions or commands that are product line specific. These will currently also be sanitized, as well as the usernames in file system paths.

Reply:
I think that this would be out-of-scope of the PR. We should re-think logging & analytics for our code-bases, though.